keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -145,26 +150,42 @@ waitForPodToBeRunning(ctx, myPod)
+#### Controller Behavior
 
-Tracing in a kubernetes cluster must be a composable component within a larger system, and allow external users or systems to define the "Root Span" that defines and bounds the scope of a trace.
+In the traditional RPC client-server tracing model, a server responds to an incoming request by performing actions including sending outgoing requests.  The server "propagates" the trace context of the incoming request by attaching it to all outgoing requests, forming a tree of requests.  In kubernetes, instead of work being driven by a single incoming request, controllers observe the state of many objects, and perform actions including writing to objects.  Unlike the traditional RPC model, a given controller action is often the result of writes from _multiple_ other controllers.  In order to form a tree from work done by controllers, a given controller action must be attributed to a single parent object.
+
+As stated in [End-User Interaction](#end-user-interaction), a trace is a graph of the **fulfilment** of user intent.  In kubernetes, the fulfilment of user intent can be restated as the process of moving objects to their desired state.  To take v1.Pod as an example, the intent of the user/component that created the pod has been fulfilled when the pod's status moves to Running.  Therefore, an object trace in kubernetes should produce a graph of the process of moving the object from its current state to its desired state.  This means **controllers should associate a given action with the object whose state would be updated by that action**.  For example, when the scheduler assigns a pod to a node, it interacts with many objects, including Pod, Node, Binding, ResourceQuota, and others.  Using the proposed model of controller tracing, the scheduler associates process of scheduling with the Pod object, since that is the object which is moved towards its desired state.  This solves our dilema above, as it gives us a rule for associating an action with a single parent object, and allows us to form a tree of controller actions.
+
+When attempting to move object `Foo` towards its desired state, a Controller must:
+
+1. Attach the trace context stored in `Foo` to all API requests. See [Tracing API Requests](#tracing-api-requests).
+1. Store the trace context of `Foo` in object `Bar` when updating the desired state of `Bar`.
+1. Export a span around work that attempts to drive the actual state of an object towards its desired state.
+1. Set the trace context of `Foo` to not be sampled when `Foo`'s status reaches the desired state.
```

Comment:
status sub-resource isnt mandatory for all resources.. e.g. the [service-account controller](https://github.com/kubernetes/kubernetes/blob/4c81edbb41f3f1a7e2d20b8ad63b6a2701421214/pkg/controller/serviceaccount/serviceaccounts_controller.go#L185) doesn't refresh its status upon "reaching desired state". how does it finish the lifespan?

Reply:
I decided that having controllers update this field is more trouble than it is worth, and removed this.  I added figuring out how to limit the lifetime of a trace to the beta criteria.